elastic/apm-server
# Revise wait_for_integration ES implementation

## Motivation/summary

Instead of polling Elasticsearch for the presence of index templates, attempt to create a data stream; this will fail if there is no index template with a matching index pattern. This decouples apm-server from the integration package, and will enable apm-server to alternatively work with the apm-data ES plugin.

Instead of eagerly checking the preconditions (which includes making these attempts to create a data stream), we now delay the checks to the first attempt to index/publish events. This means that the "publish_ready" attribute of the server info API response will always be false until the first time events are sent to apm-server.

## Checklist

~- [ ] Update [package changelog.yml](https://github.com/elastic/apm-server/blob/main/apmpackage/apm/changelog.yml) (only if changes to `apmpackage` have been made)~
~- [ ] Documentation has been updated~

## How to test these changes

1. Create an ES+Kibana stack, but don't install the integration
2. Run apm-server
3. Send some events to apm-server, confirm they are not indexed
4. Install the integration
5. Send some more events to apm-server, confirm they are (eventually) indexed

## Related issues

https://github.com/elastic/apm-server/issues/11528
